internal/provider/aws/health.go

```diff
@@ -142,8 +142,13 @@ func (r *Route53HealthCheckReconciler) Reconcile(ctx context.Context, spec provi
 func (r *Route53HealthCheckReconciler) Delete(ctx context.Context, _ *externaldns.Endpoint, probeStatus *v1alpha1.HealthCheckStatusProbe) (provider.HealthCheckResult, error) {
 	healthCheck, found, err := r.findHealthCheck(ctx, probeStatus)
 	if err != nil {
-		return provider.HealthCheckResult{}, err
+		var awserror awserr.Error
+		_ = errors.As(err, &awserror)
+		if awserror.Code() != route53.ErrCodeNoSuchHealthCheck {
+			return provider.HealthCheckResult{}, err
+		}
 	}
+
```

Comment:
What is this change for?

Reply:
fix for the bug when aws returns `NoSuchHealthCheck` error code and it is not treated well [here](https://github.com/Kuadrant/dns-operator/blob/0986c428ba16f04893e238e6016e881076b288a8/internal/controller/dnsrecord_controller.go#L89) by `client.IgnoreNotFound(err)`. It was not a problem before as we were not "validating" deletion so haven't really encountered such a scenario.